### PR TITLE
Add configurable chain separator (Issue #52 - 1/2)

### LINKED ISSUE #52: Improving QoL of usage of redis-dict
Hi @Attumm 

Lovely project. I am building atop it to have a slightly modified version. If you're open to it, I think it would be good to have these features integrated back upstream. Listing the points (in no particular order) below, please do let me know if you'd be open to contributions for the same:

1. **Making dict-behaviour more consistent, and raising errors where it can't be done**: Currently, the below code block behaves abnormally as shown below
    * Input:
    ```python
    from redis_dict import RedisDict
    
    a = RedisDict()
    a["var"]={"c":1, "c2":3}
    print("Original: ", a)
    
    a["var"]["c"] = 2
    print("Updated: ", a)
    
    a.chain_set(("var","c"),2)
    print("Updated with chain set: ", a)
    ```
    * Output:
    ```
    Original:  {'var': {'c': 1, 'c2': 3}}
    Updated:  {'var': {'c': 1, 'c2': 3}}
    Updated with chain set:  {'var': {'c': 1, 'c2': 3}, 'var:c': 2}
    ```
    * Dictionaries are treated differently depending on whether they're set directly, or using the purpose-built-functions. Hence, This could result in cases where the user believes a key should have been updated, but it never did get updated.
    * There is a difference in setting dictionaries through assignment, vs adding them using `.chain_set()`. This is non-intuitive, and ideally the user shouldn't have to worry about this.
    * Proposed behaviour:
      1. **Creation**: When the user runs `a["var"]={"c":1, "c2":3}`, it should internally created nested keys that are managed by redis (and hence updatable) using `.chain_set()` internally.
      2. **Updation**: When the user runs `a["var"]["c"] = 2`, it should raise a `NotImplementedError` that asks users to instead use `.chain_set()` instead. (I adopted this approach because of a limitation in python's language parser that doesn't allow it to uniquely identify if a nested setitem is being run, or a simple setitem. For more, can refer [here](https://stackoverflow.com/q/16676177))
2. **Supporting configurable delimiters for nested keys**:  Currently, in the `chain_xyz()` set of functions, `:` is used as a separator. However, this isn't a very scalable choice, as `:` could appear in the individual keys of the hierarchy too. Hence, this should be configurable. Additionally, the default should perhaps be a **non-ASCII** character like '➡️'

### PR DESCRIPTION
**Pull request by rsnk96**
This PR adds a configurable chain separator to RedisDict (resolves #52). Previously, the chain separator was hardcoded to `:`, which could cause key collisions when user keys contained colons. The separator is now configurable via a constructor parameter and defaults to `➡️     ` (emoji + spaces) to avoid collisions in typical use cases.

**Test code:**
```python
from redis_dict import RedisDict

# Create RedisDict with custom separator
r = RedisDict(namespace='test')
r.clear()

# Use chain operations
r.chain_set(['foo', 'bar'], 'value1')
r.chain_set(['foo', 'bar2'], 'value2')

# No collision with user keys containing ':'
r['foo:bar'] = 'direct_value'

print("Normal chain get of foo, bar: ", r.chain_get(['foo', 'bar']))
print("Specific key explicitly set:  ", r['foo:bar'])
print("Multi chain get of foo:       ", r.multi_chain_get(['foo']))
print("List of all keys:             ", list(r._scan_keys("*")))
```

**Behaviour before this PR:**
```
Normal chain get of foo, bar:  direct_value
Specific key explicitly set:   direct_value
Multi chain get of foo:        ['value2', 'direct_value']
List of all keys:              ['test:foo:bar2', 'test:foo:bar']
```

**Behaviour after this PR:**
```
Normal chain get of foo, bar:  value1
Specific key explicitly set:   direct_value
Multi chain get of foo:        ['value1', 'direct_value', 'value2']
List of all keys:              ['test:foo➡️     bar', 'test:foo:bar', 'test:foo➡️     bar2']
```

### CODE REVIEW

File: src/redis_dict/core.py [hunk 1]
```diff
@@ -12,6 +12,8 @@
 from .type_management import encoding_registry as enc_reg
 from .type_management import decoding_registry as dec_reg
 
+_DEFAULT_SEPARATOR = '➡️    '
```
Comment (Attumm) [inline src/redis_dict/core.py:15]: Since `_DEFAULT_SEPARATOR` is only used for chain operations, let's rename it to `_DEFAULT_CHAIN_SEPARATOR` to differentiate from the normal separator. 

Let's keep `':'` as the default for now, changing it would require a major version bump. We can revisit the default value in a future major release.

```suggestion
# TODO (major release): reconsider default separator value
_DEFAULT_CHAIN_SEPARATOR = ':'  
```

File: src/redis_dict/core.py [hunk 2]
```diff
@@ -47,6 +49,7 @@ def __init__(self,
              preserve_expiration: Optional[bool] = False,
              redis: "Optional[StrictRedis[Any]]" = None,
              raise_key_error_delete: bool = False,
+             separator: str = _DEFAULT_SEPARATOR,
```
Comment (Attumm) [inline src/redis_dict/core.py:52]: Since we renamed `_DEFAULT_SEPARATOR` to `_DEFAULT_CHAIN_SEPARATOR`, let's update the reference here as well. Also rename the parameter to `chain_separator` to differentiate from a regular separator.

```suggestion
            chain_separator: str = _DEFAULT_CHAIN_SEPARATOR,
```

File: src/redis_dict/core.py [hunk 3]
```diff
@@ -59,13 +62,16 @@ def __init__(self,
...
+            separator (str): Delimiter used to join keys in chain and nested-dict operations. Defaults to '➡️    '
+                , a string that cannot appear in ordinary user keys.
             **redis_kwargs (Any): Additional kwargs for Redis connection if not provided.
         """
 
         self.namespace: str = namespace
         self.expire: Union[int, timedelta, None] = expire
         self.preserve_expiration: Optional[bool] = preserve_expiration
         self.raise_key_error_delete: bool = raise_key_error_delete
+        self.separator: str = separator
```
Comment (Attumm) [inline src/redis_dict/core.py:74]: Let's rename the parameter to `chain_separator` to differentiate from a regular separator.

```suggestion
    self.chain_separator: str = _DEFAULT_CHAIN_SEPARATOR,
```

File: README.md [hunk 4]
```diff
@@ -377,6 +377,12 @@ Setup guide for configuring and utilizing encrypted Redis TLS for redis-dict.
 ### Tests
 The RedisDict library includes a comprehensive suite of tests that ensure its correctness and resilience. The test suite covers various data types, edge cases, and error handling scenarios. It also employs the Hypothesis library for property-based testing, which provides fuzz testing to evaluate the implementation
 
+Contributors can run the full test suite locally using the provided script:
+```bash
+bash scripts/tests.sh
+```
+This sets up a virtual environment (if needed) and runs all tests with verbose output.
+
```
Comment (Attumm) [inline README.md:385]: Great addition.

File: pyproject.toml [hunk 5]
```diff
@@ -120,7 +120,7 @@ exclude_lines = [
 show_missing = true
 
 [tool.mypy]
-python_version = "3.8"
+python_version = "3.9"
```
Comment (Attumm) [inline pyproject.toml:123]: There was a reason for 3.8 and not higher. Can't remember why exactly. Hmm

**Review by Attumm — CHANGES_REQUESTED** [review]

File: src/redis_dict/core.py (hunk 1 above)
Comment (rsnk96) [reply src/redis_dict/core.py:15]: Done!

File: src/redis_dict/core.py (hunk 2 above)
Comment (rsnk96) [reply src/redis_dict/core.py:52]: Done!

File: README.md (hunk 4 above)
Comment (rsnk96) [reply README.md:385]: Haha yes I figured it would help anyone trying it out!

File: src/redis_dict/core.py (hunk 3 above)
Comment (rsnk96) [reply src/redis_dict/core.py:74]: Done!

File: pyproject.toml (hunk 5 above)
Comment (rsnk96) [reply pyproject.toml:123]: Haha I'm guessing you mean the failing pylama. Well, we'll get that fixed now!
